src/Raven.Client/Documents/BulkInsert/BulkInsertOperation.cs

```diff
@@ -299,13 +299,20 @@ private class TimerState
 
         private static void HandleHeartbeat(object state)
         {
-            var timerState = (TimerState)state;
-            if (timerState.Parent.TryGetTarget(out BulkInsertOperation bulkInsert) == false)
+            try
```

Comment:
the only part here that might fail is the dispose of the timer maybe? wrap it up there?